api/users.py

```diff
@@ -52,6 +52,13 @@ def get_by_username(username):
         abort(404)
 
 
+@users.route('/users/search/', methods=['GET'])
```

Comment:
For this API I followed the REST design principle that URLs should map to resources. Search is not a resource, it is an action, so it does not really belong as an endpoint. My preference would be that searching on resources is implemented with query string parameters.